amplify-tools/amplify-tools-gradle-plugin/src/main/groovy/com/amplifyframework/tools/gradle/plugin/AmplifyTools.groovy

```diff
@@ -5,228 +5,230 @@ import groovy.json.JsonSlurper
-		project.task('addAmplifyPushToWorkspace') {
-			if(project.file('./.idea/workspace.xml').exists()) {
-				//Open file
-				def xml = new XmlParser().parse('./.idea/workspace.xml')
-				def RunManagerNode = xml.component.find { it.'@name' == 'RunManager' } as Node
-				def configAmplifyPushCheck = RunManagerNode.children().find { it.'@name' == 'amplifyPush' } as Node
-
-				if (!configAmplifyPushCheck) {
-					// Nested nodes for amplifyPush run configuration
-					def configurationNode = new Node(null, 'configuration', [name: "amplifyPush", type:"GradleRunConfiguration", factoryName:"Gradle", nameIsGenerated:"true"])
-					def externalSystemNode = new Node(configurationNode, 'ExternalSystemSettings')
-					def executionOption = new Node(externalSystemNode, 'option', [name: "executionName"])
-					def projectPathOption = new Node(externalSystemNode, 'option', [name: "externalProjectPath", value: "\$PROJECT_DIR\$"])
-					def externalSystemIdOption = new Node(externalSystemNode, 'option', [name: "externalSystemIdString", value: "GRADLE"])
-					def scriptParametersOption = new Node(externalSystemNode, 'option', [name: "scriptParameters", value: ""])
-					def taskDescriptionsOption = new Node(externalSystemNode, 'option', [name: "taskDescriptions"])
-					def descriptionList = new Node(taskDescriptionsOption, 'list')
-					def taskNamesOption = new Node(externalSystemNode, 'option', [name: "taskNames"])
-					def nameList = new Node(taskNamesOption, 'list')
-					def amplifyPushOption = new Node(nameList, 'option', [value: "amplifyPush"])
-					def vmOption = new Node(externalSystemNode, 'option', [name: "vmOptions", value: ""])
-					def systemDebugNode = new Node(configurationNode, 'GradleScriptDebugEnabled', null, true)
-					def methodNode = new Node(configurationNode, 'method', [v:"2"])
-
-					RunManagerNode.append(configurationNode)
-
-					//Save File
-					def writer = new FileWriter('./.idea/workspace.xml')
-
-					//Pretty print XML
-					groovy.xml.XmlUtil.serialize(xml, writer)
-				}
-			}
-		}
-	}
+    void apply(Project project) {
+        // profile name can be changed in amplify-gradle-config
+        def profile = 'default'
+        def accessKeyId = null
+        def secretAccessKey = null
+        def region = null
+        def envName = 'amplify'
+        def syncEnabled = 'true'
+        def gradleConfigFileName = 'amplify-gradle-config.json'
+
+        project.task('createAmplifyApp') {
+            def npx = 'npx'
+
+            if (project.file(gradleConfigFileName).exists()) {
+                return
+            }
+
+            if (Os.isFamily(Os.FAMILY_WINDOWS)) {
+                npx += '.cmd'
+            }
+
+            try {
+                project.exec {
+                    commandLine npx, 'amplify-app', '--platform', 'android'
+                }
+            } catch (commandLineFailure) {
+                throw new Exception('Node.js is not installed. Visit https://nodejs.org/en/download/ to install it.')
```

Comment:
Maybe we should say that npx was not found instead since node could be installed without npx

Reply:
Hm, I don't agree, because most Android developers don't know what npx is. We have a better shot if we say node.js at comprehension.

Reply:
Does npx come bundled with node? I thought my laptop had been in a state at one point where I had `node`, but not `npx`?

Reply:
npx comes bundled with npm (after version 5.something), which comes bundled with node. In Node 10+, all users who have node installed should also have npx installed. I tested this on a new system with Windows, installed Node 12 installed, and have npx available.